code/datums/weather/weather_types/rain.dm

```diff
@@ -86,10 +86,15 @@
 			if(H.belt && wash_obj(H.belt))
 				H.update_inv_belt()
 
-/datum/weather/rain/weather_act_turf(turf/T)
-	for(var/O in T) //Clean cleanable decals in affected areas
-		if(is_cleanable(O))
-			qdel(O)
+/datum/weather/rain/weather_act_turf(turf/open/T)
+	var/cleaned
+	if(!cleaned)
+		for(var/obj/effect/decal/O in T) //Clean cleanable decals in affected areas
+			if(is_cleanable(O))
+				qdel(O)
+				cleaned = 1
+				CHECK_TICK
```

Comment:
Since you're breaking out of the loop, a CHECK_TICK here is kind of nonsensical. It should instead be done in whatever calls `weather_act_turf`.